tests/test_bms_multi_model_output.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+import pytest
+from theorist.bms import Tree
+
+from autora.skl.bms import BMSRegressor
+
+
+@pytest.fixture
+def curve_to_fit():
+    x = np.linspace(-10, 10, 100).reshape(-1, 1)
+    y = (x**3.0) + (2.0 * x**2.0) + (17.0 * x) - 1
+    return x, y
+
+
+def test_bms_models(curve_to_fit):
+    x, y = curve_to_fit
+    regressor = BMSRegressor(epochs=100)
+
+    regressor.fit(x, y)
+
+    print(regressor.models_)
+
+    assert len(regressor.models_) == 20  # Currently hardcoded
```

Comment:
Would it be possible to determine the number of models in advance, e.g., via an input argument, and not hardcode it? If so, I'm wondering if it might be better to test whether the number of models matches the input argument...

Reply:
We could definitely do that. Currently, the number of models corresponds to the number of temperatures you specify when initializing the ```BMSRegressor``` - something which Ben is currently making more clear in the documentation #170 . We could easily add an input argument to return a determined number of models if that number is <=20, if you would like. Being able to return more than 20 is also easy to do, but then becomes a discussion about temperatures and BMS implementation.

Reply:
So with that in mind, we can remove the hard-coding in this file at least with this commit (below)

Reply:
aaaah, that's the thing – then let's not hard code it, let's just make sure that the length in the test case is the same as the number of temperatures [using `len(temperatures)` or whatever to get the number of temperatures we care about]

Reply:
... which you did already. Nice!